### ISSUE TITLE
4AGE 20V ROM

### ISSUE THREAD

**Issue by digitaltrousers** [jonsole/toyotune#2]
Hi there. I remember back in the day following your thread on one of the MR2 forums (which one was it again?)
I remember you extracted the ROM from a 4AGE 20V ECU. Is it still accessible anywhere, or is it possible to post it in this repository?
Cheers,
Dan.

**Comment by jonsole** [comment]
Hi,  I've upload the 2 4A-GE ROMs that I have.  I only kept a record of the chip numbers in the ECUs, so I don't know for sure if either of them is for the 20v ECU.  The one labelled D151803-7990 is from the later chip so that's the most likely one.

**Comment by digitaltrousers** [comment]
Thanks. As far as the 20V's, my silvertop chip is D151803-0360, and I believe D151804-8081 is (one of) the blacktop chip(s). I guess there are different revisions and differences between auto and manual trans, different regions, etc.

**Comment by jonsole** [comment]
I'm now 100% sure D151803-7990 is for some 4A-GE 20v variant as I found some old messages from the twobrutal forum. 
I was sent the ECU to fix leaking caps and to see if the ROM daughterboard would fit.  It didn't but I extracted the ROM image whilst I had the ECU.  It was a Silvertop ECU.

This http://www.woodsport.org/forum/showthread.php?15236-ECU-Code-Disassembly&p=244535&viewfull=1#post244535 might jog your memory :-)
